File: src/main/webapp/js/modulosPagos/Deuda.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function Deuda(colegiad, tipoPago, detalleDeuda, deudaTotal, periodo){
    this.colegiad = colegiad;
    this.tipoPago = tipoPago;
    this.periodo = periodo;
    this.detalleDeuda = detalleDeuda;
    this.deudaTotal = deudaTotal;
    this.indError = "";
    this.msjError = "";
    this.listarDeuda = listarDeuda;
    this.mantDeuda = mantDeuda;
}

function DeudaDetalle(tipoPago, periodo, deudaTotal){
    this.tipoPago = tipoPago;
    this.periodo = periodo;
    this.deudaTotal = deudaTotal;
    this.indError = "";
    this.msjError = "";
}

function listarDeuda(tipoOperacion){
    var deudaRep;
    var colegiado = {
        "dni" : this.colegiad.dni,
        "numColegiatura" : this.colegiad.numColegiatura
    };
    var deuda ={
        "colegiad": colegiado,
        "tipoPago": this.tipoPago
    };
    deudaRep = new Deuda();
    $.ajax({
        url : "../pagos/listarDeuda.action",
        type : "POST",
        contentType : 'application/json; charset=utf-8',
        dataType : 'json',
        async: false,
        mimeType: 'application/json',
        data : JSON.stringify({tipoOperacion: tipoOperacion, deuda : deuda}),
        success : function(data) {
            deudaRep = data;
        }
    });
    return deudaRep;
}

function mantDeuda(tipoOperacion){
    var deudaRep;

    var deuda ={
        "colegiad": this.colegiad,
        "tipoPago": this.tipoPago,
        "periodo": this.periodo
    };
    deudaRep = new Deuda();
    $.ajax({
        url : "../pagos/mantDeuda.action",
        type : "POST",
        contentType : 'application/json; charset=utf-8',
        dataType : 'json',
        async: false,
        mimeType: 'application/json',
        data : JSON.stringify({tipoOperacion: tipoOperacion, deuda : deuda}),
        success : function(data) {
            deudaRep = data;
        }
    });
    return deudaRep;
}

File: src/main/java/com/whnm/sicqfdp/beans/Deuda.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.whnm.sicqfdp.beans;

import java.util.List;

/**
 *
 * @author whnm
 */
public class Deuda {
    private Colegiado colegiad;
    private TipoPago tipoPago;
    private String periodo;
    private Double deudaTotal;
    private List<DetalleDeuda> detalleDeuda;
    private Integer indError;
    private String msjError;

    public Colegiado getColegiad() {
        return colegiad;
    }

    public void setColegiad(Colegiado colegiad) {
        this.colegiad = colegiad;
    }

    public TipoPago getTipoPago() {
        return tipoPago;
    }

    public void setTipoPago(TipoPago tipoPago) {
        this.tipoPago = tipoPago;
    }

    public Double getDeudaTotal() {
        return deudaTotal;
    }

    public void setDeudaTotal(Double deudaTotal) {
        this.deudaTotal = deudaTotal;
    }

    public List<DetalleDeuda> getDetalleDeuda() {
        return detalleDeuda;
    }

    public void setDetalleDeuda(List<DetalleDeuda> detalleDeuda) {
        this.detalleDeuda = detalleDeuda;
    }

    public Integer getIndError() {
        return indError;
    }

    public void setIndError(Integer indError) {
        this.indError = indError;
    }

    public String getMsjError() {
        return msjError;
    }

    public void setMsjError(String msjError) {
        this.msjError = msjError;
    }

    public String getPeriodo() {
        return periodo;
    }

    public void setPeriodo(String periodo) {
        this.periodo = periodo;
    }
}
